### PR TITLE
Add migration guide for page apis

### PR DESCRIPTION
**Pull request by chunhtai**
migration guide for https://github.com/flutter/flutter/pull/137792

## Presubmit checklist

- [ ] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [ ] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [ ] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

**Review by sfshaza2 — COMMENTED** [review]
Can you add this breaking change to the index file?

File: src/content/release/breaking-changes/navigator-and-page-api.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
+---
+title: Navigator's page APIs breaking change
+description: >
+  Changes to Navigator's page APIs.
```
Comment (parlough) [inline src/content/release/breaking-changes/navigator-and-page-api.md:4]: Could you add a word or two to be more specific here, as the APIs might have other breaking changes in the future.

Comment (sfshaza2) [reply src/content/release/breaking-changes/navigator-and-page-api.md:4]: good point

**Comment by chunhtai** [comment]
all done

**Review by sfshaza2 — APPROVED** [review]
lgtm

**Comment by aliyazdi75** [comment]
Hi @chunhtai,

Since this api is deperecated, I think it's better to mention to use `onPopInvoked` to use the result, something like this:
```dart
class CustomAppPage<T> extends Page<T> {
  const CustomAppPage({
    super.key,
    super.onPopInvoked,
    required this.child,
  });

  final Widget child;
}

CustomAppPage(
    key: ValueKey(route),
    onPopInvoked: (didPop, result) =>
        _routerState.popResult.complete(result),
    child: CustomPage(),
),
```

**Pull request merged**